File: python/shroomdk/models/compass/core/sql_statement.py
from typing import Any, Dict, List, Optional

from pydantic import BaseModel

from .column_metadata import ColumnMetadata
from .tags import Tags


class SqlStatement(BaseModel):
    id: str
    statementHash: str
    sql: str
    columnMetadata: Optional[ColumnMetadata]
    userId: str
    tags: Tags
    createdAt: str
    updatedAt: str

File: js/src/types/compass/core/sql-statement.ts
import { ColumnMetadata } from "./column-metadata";
import { Tags } from "./tags";

export interface SqlStatement {
  id: string;
  statementHash: string;
  sql: string;
  columnMetadata: ColumnMetadata | null;
  userId: string;
  tags: Tags;
  createdAt: string;
  updatedAt: string;
}
